GitUI/CommandsDialogs/FormFileHistory.cs

```diff
@@ -226,7 +226,12 @@ private void LoadFileHistory()
 
             ObjectId objectId = rev.IsArtificial ? RevisionGrid.CurrentCheckout : rev.ObjectId;
 
-            return RevisionGrid.FilePathByObjectId.TryGetValue(objectId, out string? path) ? path : null;
+            if (!RevisionGrid.FilePathByObjectId.TryGetValue(objectId, out string? path))
+            {
+                path = RevisionGrid.GetRevisionFileName(FileName, objectId.ToString());
+            }
+
+            return path;
```

Comment:
This feels like we're leaking implementation details. Can we refactor it so that the caller has to do a single call?
```suggestion
            return RevisionGrid.GetRevisionFileName(FileName, objectId);
```